server/service/apple_mdm_test.go

```diff
@@ -1804,6 +1818,115 @@ func TestGenerateEnrollmentProfileMobileConfig(t *testing.T) {
 	require.Contains(t, string(b), "foo&amp;bar")
 }
 
+func TestEnsureFleetdConfig(t *testing.T) {
+	testError := errors.New("test error")
+	testURL := "https://example.com"
+	testTeamName := "test-team"
+	testEnrollSecret := "xyz"
+
+	t.Run("no enroll secret found", func(t *testing.T) {
+		ctx := context.Background()
+		ds := new(mock.Store)
+
+		ds.AppConfigFunc = func(ctx context.Context) (*fleet.AppConfig, error) {
+			return &fleet.AppConfig{}, nil
+		}
+		ds.GetEnrollSecretsFunc = func(ctx context.Context, teamID *uint) ([]*fleet.EnrollSecret, error) {
+			return []*fleet.EnrollSecret{}, nil
+		}
+		err := ensureFleetdConfig(ctx, ds)
+		require.ErrorContains(t, err, "unable to find an enroll secret")
+	})
+
+	t.Run("error when the default team doesn't exist", func(t *testing.T) {
```

Comment:
Nit, but I think this tests when it fails to get the default team, not that it doesn't exist (in which case it'd create a global enrollment profile)?